### PR TITLE
Vendor libmaxminddb

### PR DESCRIPTION
**Pull request by josh**
For #1.

Just hacked the basics together to get things working. Traveling right now.
- [x] Using 0.5.3 since thats what I was using from homebrew. But it looks like the latest is now 1.0.3. We should probably switch to that. At least this gem can normalize the library version everyone is using.
- [x] Should a tar just be checked in or is the 100 loose files okay?
- [x] Whats the conventional place to "install" the libmaxminddb lib here?
- [x] `rake compile` leaves a ton of untracked files in the working directory
- [x] Is there anyway to avoid the additional libmaxminddb object and statically build out a single geoip2_compat object?
- [ ] Should we get CI setup on this thing?

cc @dbussink @brianmario

### CODE REVIEW

**Comment by dbussink** [comment]
> Using 0.5.3 since thats what I was using from homebrew. But it looks like the latest is now 1.0.3. We should probably switch to that. At least this gem can normalize the library version everyone is using.

We should definitely use 1.0.3 since we run that's the latest version we also run in production.

**Comment by josh** [comment]
Upgraded to 1.0.3.

Also switched to only tracking the relevant source and header files and its statically built.

File: Rakefile [hunk 1]
```diff
@@ -18,4 +18,17 @@ task :download do
   end
 end
 
+task :vendor do
+  version = "1.0.3"
```
Comment (josh) [inline Rakefile:22]: This version is duplicated a few times. Think it should be a ruby const somehow?

**Comment by dbussink** [comment]
Going to merge this and see if I can get CI up and running

**Comment by josh** [comment]
:metal:

**Pull request merged**